dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace FunctionCalling;
+
+/// <summary>
+/// These samples demonstrate how to advertise functions to AI model based on a context.
+/// </summary>
+public class ContextDependentAdvertising(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// This sample demonstrates how to advertise functions to AI model based on the context of the chat history.
+    /// It advertise functions to the AI model based on the game sate.
+    /// For example, if the maze has not been created, advertise the create maze function only to prevent the AI model
+    /// from adding traps or treasures to the maze before it is created.
+    /// </summary>
+    [Fact]
+    public async Task AdvertiseFunctionsDependingOnContextPerUserInteractionAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        IChatCompletionService chatCompletionService = kernel.GetRequiredService<IChatCompletionService>();
+
+        // Tracking number of iterations to avoid infinite loop.
+        int maxIteration = 10;
+        int iteration = 0;
+
+        // Define the functions for AI model to call.
+        var gameUtils = kernel.ImportPluginFromType<GameUtils>();
+        KernelFunction createMaze = gameUtils["CreateMaze"];
+        KernelFunction addTraps = gameUtils["AddTrapsToMaze"];
+        KernelFunction addTreasures = gameUtils["AddTreasuresToMaze"];
+        KernelFunction playGame = gameUtils["PlayGame"];
+
+        ChatHistory chatHistory = new();
```

Comment:
nit.

If we added a implicit `string` operator to our `ChatMessageContent` that imply a `User` role with the `string` as `Content` property we could do similar to the `OpenAI SDK`, which is handy. `ChatHistory chatHistory = ["I would like to play..."]`

```suggestion
        ChatHistory chatHistory = [];
```